utils/cputils.py

```diff
@@ -0,0 +1,153 @@
+            sqlite_file_path.name, f"{name}.sqlite"
+        )
+
+        # change the file name in the directory
+        pathlib.Path(sqlite_file_path).rename(pathlib.Path(new_file_name))
+        print(f"The file is renamed to {pathlib.Path(new_file_name).name}!")
+
+    except FileNotFoundError as e:
+        print(
+            f"The NF1_data.sqlite file is not found in directory. Either the pipeline wasn't ran properly or the file is already renamed.\n"
+            f"{e}"
+        )
+
+
+def run_cellprofiler(
+    path_to_pipeline: str,
+    path_to_output: str,
+    path_to_images: str,
+    sqlite_name: Optional[None | str] = None,
+    analysis_run: Optional[False | bool] = False,
+):
+    """Run CellProfiler on data. It can be used for either a illumination correction pipeline (default) and analysis pipeline (when
+    parameter is set to True).
+
+    Args:
+        path_to_pipeline (str): path to the CellProfiler .cppipe file with the segmentation and feature measurement modules
+        path_to_output (str): path to the output folder
+        path_to_images (str): path to the images
+        sqlite_name (str, optional): string with name for SQLite file for an analysis pipeline if you plan on running
+        multiple sets of images (e.g., per plate) so that the outputs will have different names (default is None)
+        analysis_run (bool, optional): will use functions to complete an analysis pipeline (default is False)
+    """
+    # check to make sure paths to pipeline and directory of images are correct before running the pipeline
+    if not pathlib.Path(path_to_pipeline):
+        raise FileNotFoundError(f"The file '{pathlib.Path(path_to_pipeline).name}' does not exist")
+    if not pathlib.Path(path_to_images).is_dir():
+        raise FileNotFoundError(f"Directory '{pathlib.Path(path_to_images).name}' does not exist or is not a directory")
+
+    # make logs directory
+    log_dir = pathlib.Path("./logs")
+    os.makedirs(log_dir, exist_ok=True)
+
+    # make output directory if it is not already created
+    os.makedirs(path_to_output, exist_ok=True)
+
+    # run CellProfiler illumination correction pipeline
+    if not analysis_run:
+        print(f"Starting CellProfiler run on {pathlib.Path(path_to_images).name}")
+        # a log file is created for each plate or data set name (based on folder name with images) that holds all outputs and errors
+        with open(
+            pathlib.Path(
+                f"logs/cellprofiler_output_{pathlib.Path(path_to_images).name}.log"
+            ),
+            "w",
+        ) as cellprofiler_output_file:
+            # run CellProfiler for a illumination correction pipeline
+            command = [
+                "cellprofiler",
+                "-c",
+                "-r",
+                "-p",
+                path_to_pipeline,
+                "-o",
+                path_to_output,
+                "-i",
+                path_to_images,
+            ]
```

Comment:
consider writing some documentation what these parameters do. You can provide a link to the documentation where these parameters are listed as a single comment line.